src/lib/backoff.ts

```diff
@@ -0,0 +1,88 @@
+import { promisify } from 'util';
+
+const sleep = promisify(setTimeout);
+
+export type BackoffCallback = {
```

Comment:
`OnRetryCallback` ? Since `backoff` is in the name of the module?

Reply:
`OnRetry` might be enough too

Reply:
Actually the type does not point to a callback so `OnRetryOpts` might make more sense